ext/session/session.c

```diff
@@ -369,20 +373,19 @@ PHPAPI zend_result php_session_valid_key(const char *key) /* {{{ */
 				|| (c >= '0' && c <= '9')
 				|| c == ','
 				|| c == '-')) {
-			ret = FAILURE;
-			break;
+			return FAILURE;
 		}
 	}
 
 	len = p - key;
 
 	/* Somewhat arbitrary length limit here, but should be way more than
 	   anyone needs and avoids file-level warnings later on if we exceed MAX_PATH */
-	if (len == 0 || len > PS_MAX_SID_LENGTH) {
-		ret = FAILURE;
+	if (len > PS_MAX_SID_LENGTH) {
```

Comment:
There should be an assertion that the string length is not 0 if you remove the condition.

Reply:
This condition didn't work. Empty strings are currently allowed in this function so I just removed this unnecessary condition. https://3v4l.org/g4Skm

So adding this check would violate BC. In my opinion, I don't see the point why empty string shouldn't be allowed. It doesn't contain forbidden characters.

Reply:
I reverted this change. The empty key is still allowed as the input of `session_create_id()`. The second use of this function in `mod_files.c` will throw an exception as it currently works.